.storybook/stories/DataFiltering/components/DashboardPagination.tsx

```diff
@@ -38,25 +105,119 @@ export const DashboardPagination = () => {
 
 	if (!data.length) return null;
 
+	const hasChangedColumnSettings = !Object.values(
+		formState.activeColumns
+	).every(Boolean);
+	const hasChangedItemsPerPageSettings = pagination.perPage !== 10;
+	const appliedSettingsCount =
+		Number(hasChangedColumnSettings) + Number(hasChangedItemsPerPageSettings);
+
 	return (
-		<PaginationButtons
-			currentPage={displayText.page}
-			onChange={(page) => setPagination({ perPage: displayText.perPage, page })}
-			totalPages={totalPages}
-			itemRangeText={displayText.itemRangeText}
-			itemsPerPage={displayText.perPage}
-			onItemsPerPageChange={(perPage) =>
-				setPagination({
-					page: getValidPage({
-						currentPage: pagination.page,
-						totalItems,
-						perPage,
-						prevPerPage: pagination.perPage,
-					}),
-					perPage,
-				})
-			}
-		/>
+		<>
+			<Box
+				display="grid"
+				gap={1}
+				css={{
+					gridTemplateColumns: '1fr',
+					[tokens.mediaQuery.min.lg]: {
+						gridTemplateColumns: '2fr 1fr',
+					},
+				}}
+			>
+				<PaginationButtons
+					currentPage={displayText.page}
+					onChange={(page) =>
+						setPagination({ perPage: displayText.perPage, page })
+					}
+					totalPages={totalPages}
+					itemRangeText={displayText.itemRangeText}
+				/>
+
+				<Button
+					css={{
+						[tokens.mediaQuery.min.lg]: {
+							justifySelf: 'end',
+						},
+					}}
+					iconBefore={SettingsIcon}
+					onClick={openDrawer}
+					variant="tertiary"
+				>
+					Table settings ({appliedSettingsCount})
+				</Button>
+			</Box>
+
+			<Drawer
+				actions={
+					<ButtonGroup>
+						<Button
+							type="submit"
+							form="table-settings-form-id"
+							onClick={onSubmitForm}
+						>
+							Save settings
+						</Button>
+
+						<Button
+							type="reset"
+							form="table-settings-form-id"
+							onClick={() => console.log('Reset table settings form')}
```

Comment:
Do we want this to function?

Reply:
Oh yeah, good one.

- [ ] reset settings button